core/src/main/kotlin/com/justai/jaicf/reactions/Reactions.kt

```diff
@@ -129,20 +129,32 @@ abstract class Reactions {
      *
      * @return [AudioReaction] for logging purposes
      * */
-    open fun audio(url: String): AudioReaction = createAudioReaction(url)
+    open fun audio(url: String): AudioReaction = AudioReaction(url, currentState)
 
-    protected fun createAudioReaction(audioUrl: String) =
-        AudioReaction(audioUrl, botContext.dialogContext.currentState, loggingContext)
+    protected fun registerReaction(reaction: Reaction) {
+        loggingContext.reactions.add(reaction)
+    }
+
+    protected fun SayReaction.Companion.create(text: String) =
+        SayReaction(text, currentState).also {
```

Comment:
Do we actually need `currentState` in `Reaction` object as `Reactions` is always aware of the current state and can use it on its own in `registerReaction`?

Reply:
Well, this will mean that Reaction.fromState will be either var, or reaction will be splitted two variables (one with state, second with reaction itself). 
I don't think we need to split it right now, as we always have some state which was invoking reaction, meaning that state is a part of Reaction object. (At least it sounds ok for me)